scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+
+def _load_dataset(fils):
+    if len(fils) == 0:
+        print("Input file list is empty.")
+        return None
+    elif len(fils) > 1:
+        return xr.open_mfdataset(fils, combine='by_coords')
+    else:
+        sfil = str(fils[0])
+        return xr.open_dataset(sfil)
+    #End if
+#End def
+
+########
+
+def _data_calcs(var, ts_ds, subset=None):
+    """
+    Function to calculate global weighted data
+    ----
+        - weight global data by latitude
+        - option for lat/lon subset
+        - returns
+            * month lengths
+            * climo years
+            * units
+    """
+
+    time = ts_ds['time']
+    time = xr.DataArray(ts_ds['time_bnds'].load().mean(dim='nbnd').values, dims=time.dims, attrs=time.attrs)
+    ts_ds['time'] = time
+    ts_ds.assign_coords(time=time)
+    ts_ds = xr.decode_cf(ts_ds)
+
+    if subset != None:
+        ts_ds = ts_ds.sel(lat=slice(subset["s"],subset["n"]), lon=slice(subset["w"],subset["e"])) 
+
+    data = ts_ds[var].squeeze()
+    month_length = data.time.dt.days_in_month
+    unit = data.units
+
+    #global weighting
+    w = np.cos(np.radians(data.lat))
+    avg = data.weighted(w).mean(dim=("lat","lon"))
+
+    yrs = np.unique([str(val.item().timetuple().tm_year).zfill(4) for _,val in enumerate(ts_ds["time"])])
+
+    return avg,month_length,yrs,unit
+
+########
+
+def seasonal_data(data, month_length):
+    """
+    Function to grab seasonal weighted data, grouped by season
+    -----
+        - DJF, MAM, JJA, and SON
+    """
+
+    weighted_mean = ((data * month_length).resample(time='QS-DEC').sum() /
+                          month_length.resample(time='QS-DEC').sum())
+
+    mdata_seasonal_mean = weighted_mean.groupby('time.season')
+
+    return mdata_seasonal_mean
+
+########
+
+def _get_seasonal_data(ts_var_list, all_case_names, case_ts_locs):
+    """
+    Gather seasonally weighted data
+    -----
+        - 
+    """
+
+    #Keep track of variables that have vertical levels for now
+    #There is probably a better way to ignore these vars - JR
+    del_s = []
+
+    vals = OrderedDict()
+    yrs = {}
+    units = {}
+    for var in ts_var_list:
+        if var not in vals:
+            vals[var] = OrderedDict()
+        for case_idx, case_name in enumerate(all_case_names):
+            fils = sorted(list(Path(case_ts_locs[case_idx]).glob(f"*{var}.*.nc")))
+            ts_ds = _load_dataset(fils)
+
+            #Check if variable has a vertical coordinate:
+            if 'lev' in ts_ds.coords or 'ilev' in ts_ds.coords:
+                print(f"\t   Variable '{var}' has a vertical dimension, "+\
+                    "which is currently not supported for the time series plot. Skipping...")
+
+                if var in ts_var_list:
+                    #Keep track of vars with levels
+                    #Probably a better way to ignore variables
+                    #with levels than this.
+                    #Look into it - JR
+                    del_s.append(var)
```

Comment:
If you convert this to a set, then you'll need to change this to:

```suggestion
                    del_s.add(var)
```